Lib/test/test_importlib/import_/test_api.py

```diff
@@ -82,6 +82,20 @@ def test_fromlist_load_error_propagates(self):
                     self.__import__(PKG_NAME,
                                     fromlist=[SUBMOD_NAME.rpartition('.')[-1]])
 
+    def test_blocked_fromlist(self):
+        # If fromlist entry is None, let a ModuleNotFoundError propagate.
+        # issue31642
+        mod = types.ModuleType(PKG_NAME)
+        mod.__path__ = ['XXX']
```

Comment:
You can actually just set `__path__` to `[]` and it will still count as a package.